src/types/types.d.ts

```diff
@@ -344,14 +344,60 @@ export enum Sep8ApprovalStatus {
   SUCCESS = "success",
 }
 
+export enum Sep8Step {
+  DISABLED = "disabled",
+  STARTING = "starting",
+  PENDING = "pending",
+  TRANSACTION_REVISED = "transaction_revised",
+  ACTION_REQUIRED = "action_required",
+  SENT_ACTION_REQUIRED_FIELDS = "sent_action_required_fields",
+  COMPLETE = "complete",
+}
+
 export interface Sep8PaymentTransactionParams extends PaymentTransactionParams {
   approvalServer: string;
   assetCode: string;
   assetIssuer: string;
 }
 
+/**
+ * Returns the next SEP-8 step to be displayed after the current step succeeds.
+ * @param {Sep8Step} currentStep the current SEP-8 step.
+ * @param {Sep8ApprovalStatus} [approvalStatus] the approval status returned from the SEP-8 server. This value will only be used if `currentStep === Sep8Step.STARTING`.
+ * @returns {Sep8Step} the next SEP-8 step for the application.
+ */
+export const Sep8NextStepOnSuccess = ({
+  currentStep,
+  approvalStatus,
+}: {
+  currentStep: Sep8Step;
+  approvalStatus?: Sep8ApprovalStatus;
+}): Sep8Step => {
+  const nextStepDict: { [key: Sep8Step]: Sep8Step } = {
+    [Sep8Step.DISABLED]: Sep8Step.STARTING,
+    [Sep8Step.STARTING]: (() => {
+      const approvalStatusDict: { [key: Sep8ApprovalStatus]: Sep8Step } = {
+        [Sep8ApprovalStatus.ACTION_REQUIRED]: Sep8Step.ACTION_REQUIRED,
+        [Sep8ApprovalStatus.PENDING]: Sep8Step.PENDING,
+        [Sep8ApprovalStatus.REVISED]: Sep8Step.TRANSACTION_REVISED,
+        [Sep8ApprovalStatus.SUCCESS]: Sep8Step.TRANSACTION_REVISED,
+        [Sep8ApprovalStatus.REJECTED]: currentStep,
+      };
+      return approvalStatusDict[approvalStatus];
+    })(),
+    [Sep8Step.PENDING]: Sep8Step.DISABLED,
+    [Sep8Step.TRANSACTION_REVISED]: Sep8Step.COMPLETE,
+    [Sep8Step.ACTION_REQUIRED]: Sep8Step.SENT_ACTION_REQUIRED_FIELDS,
+    [Sep8Step.SENT_ACTION_REQUIRED_FIELDS]: Sep8Step.STARTING,
+    [Sep8Step.COMPLETE]: Sep8Step.DISABLED,
+  };
+
+  return nextStepDict[currentStep];
+};
+
```

Comment:
This seems more like a helper function than a type, maybe we could move it to `helpers`?

Reply:
Great suggestion, thanks for highlighting that 😄 
Done in https://github.com/stellar/stellar-demo-wallet/pull/164/commits/bee56e5af78afdb0135e762b4b3bd072640b74df